cmd/run/run.go

```diff
@@ -327,7 +329,13 @@ func (s *ServerContext) Run(ctx context.Context, config *serverconfig.Config) er
 		sqlcommon.WithMaxIdleConns(config.Datastore.MaxIdleConns),
 		sqlcommon.WithConnMaxIdleTime(config.Datastore.ConnMaxIdleTime),
 		sqlcommon.WithConnMaxLifetime(config.Datastore.ConnMaxLifetime),
-	)
+	}
+
+	if config.Datastore.Metrics.Enabled {
+		datastoreOptions = append(datastoreOptions, sqlcommon.WithMetrics())
+	}
```

Comment:
There is a test called `TestServerMetricsReporting`. Can you turn on this flag in that test and then assert that one of the metrics is emitted?

Reply:
Ah... that test is using the `memory` datastore. We'd need to change it to spin up a Postgres datastore. Hmmm

Reply:
Let's do that in a follow-up PR